lib/evmone/instructions.hpp

```diff
@@ -701,6 +701,43 @@ inline code_iterator jumpi(StackTop stack, ExecutionState& state, code_iterator
     return cond ? jump_impl(state, dst) : pos + 1;
 }
 
+inline code_iterator rjump(StackTop /*stack*/, ExecutionState& /*state*/, code_iterator pc) noexcept
+{
+    // Reading next 2 bytes is guaranteed to be safe by deploy-time validation.
+    const auto offset_hi = pc[1];
+    const auto offset_lo = pc[2];
+    const auto offset = static_cast<int16_t>((offset_hi << 8) + offset_lo);
+    return pc + 3 + offset;  // PC_post_rjump + offset
+}
+
+inline code_iterator rjumpi(StackTop stack, ExecutionState& state, code_iterator pc) noexcept
+{
+    const auto cond = stack.pop();
+    return cond ? rjump(stack, state, pc) : pc + 3;
+}
+
+inline code_iterator rjumpv(StackTop stack, ExecutionState& /*state*/, code_iterator pc) noexcept
+{
+    constexpr auto REL_OFFSET_SIZE = sizeof(int16_t);
+    const auto case_ = stack.pop();
+
+    const auto count = pc[1];
+    const auto pc_post = pc + 1 + 1 /* count */ + count * REL_OFFSET_SIZE /* tbl */;
+
+    if (case_ >= count)
+    {
+        return pc_post;
+    }
+    else
+    {
+        const auto rel_offset_hi = pc[1 + static_cast<uint16_t>(case_) * REL_OFFSET_SIZE];
+        const auto rel_offset_lo = pc[2 + static_cast<uint16_t>(case_) * REL_OFFSET_SIZE];
```

Comment:
Is this off-by-one error? 
```suggestion
        const auto rel_offset_hi = pc[2 + static_cast<uint16_t>(case_) * REL_OFFSET_SIZE];
        const auto rel_offset_lo = pc[3 + static_cast<uint16_t>(case_) * REL_OFFSET_SIZE];
```
`pc[1]` is count, for `case == 0`, it should take `pc[2]` and `pc[3]`.